pkg/provider/aci.go

```diff
@@ -298,6 +298,112 @@ func addAzureAttributes(ctx context.Context, span trace.Span, p *ACIProvider) co
 	})
 }
 
+func (p *ACIProvider) CreatePodData(ctx context.Context, pod *v1.Pod, secretsMap map[string]v1.Secret, configMaps map[string]v1.ConfigMap) (*azaciv2.ContainerGroup, error) {
+	var err error
+	ctx, span := trace.StartSpan(ctx, "aci.CreatePod")
+	defer span.End()
+	ctx = addAzureAttributes(ctx, span, p)
+
+	cg := &azaciv2.ContainerGroup{
+		Properties: &azaciv2.ContainerGroupPropertiesProperties{},
+	}
+
+	os := azaciv2.OperatingSystemTypes(p.operatingSystem)
+	policy := azaciv2.ContainerGroupRestartPolicy(pod.Spec.RestartPolicy)
+
+	cg.Location = &p.region
+	cg.Properties.RestartPolicy = &policy
+	cg.Properties.OSType = &os
+
+	// get containers
+	containers, err := p.getContainers(pod)
+	if err != nil {
+		return nil, err
```

Comment:
nit: maybe better error handling can be added throughout `CreatePodData`. wrapping these `errs` from these conditionals with additional context could be one way to do that.